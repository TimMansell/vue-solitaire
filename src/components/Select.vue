<template>
  <div class="select">
    <label class="select__label" :for="id">{{ label }}:</label>
    <select
      class="select__dropdown"
      :id="id"
      :value="modelValue"
      @change="$emit('update:modelValue', $event.target.value)"
      data-test="select"
    >
      <option v-for="(item, index) in items" :value="item.value" :key="index">
        {{ item.text }}
      </option>
    </select>
  </div>
</template>

<script>
export default {
  name: 'Select',
  props: {
    label: {
      type: String,
      default: '',
    },
    modelValue: {
      type: [String, Number],
      default: '',
    },
    items: {
      type: Array,
      default: () => [],
    },
  },
  emits: ['update:modelValue'],
  computed: {
    id() {
      const { label } = this;

      return label.split(' ').join('-');
    },
  },
};
</script>

<style scoped lang="scss">
.select {
  &__label {
    margin-right: var(--mg-sm);
  }

  &__dropdown {
    padding: var(--pd-xs);
    border: 1px solid var(--bdr-primary);
    border-radius: var(--bdr-radius-md);
    background: var(--bg-secondary);
  }
}
</style>
