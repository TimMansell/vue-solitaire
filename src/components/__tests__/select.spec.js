import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import Select from '@/components/Select.vue';

describe('Select.vue', () => {
  it('renders the component without crashing - number', () => {
    const wrapper = shallowMount(Select, {
      props: {
        label: 'Label 1',
        value: 1,
        items: [1, 2, 3],
      },
    });

    expect(wrapper.isVisible()).toBe(true);
  });

  it('renders the component without crashing - string', () => {
    const wrapper = shallowMount(Select, {
      props: {
        label: 'Label 1',
        value: 'value 2',
        items: ['value 1', 'value 2', 'value 3'],
      },
    });

    expect(wrapper.isVisible()).toBe(true);
  });

  it('renders formats id from label correctly', () => {
    const wrapper = shallowMount(Select, {
      props: {
        label: 'Label 1',
      },
    });

    expect(wrapper.find('[data-test="select"]').attributes('id')).toBe(
      'Label-1'
    );
  });
});
