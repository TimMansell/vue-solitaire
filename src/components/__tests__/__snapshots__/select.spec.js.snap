// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select.vue matches snapshot - number 1`] = `
<div class="select"><label for="Label-1" class="select__label">Label 1:</label> <select id="Label-1" data-test="select" class="select__dropdown">
    <option value="">

    </option>
    <option value="">

    </option>
    <option value="">

    </option>
  </select></div>
`;

exports[`Select.vue matches snapshot - string 1`] = `
<div class="select"><label for="Label-1" class="select__label">Label 1:</label> <select id="Label-1" data-test="select" class="select__dropdown">
    <option value="">

    </option>
    <option value="">

    </option>
    <option value="">

    </option>
  </select></div>
`;
